src/audio/google/google_rtc_audio_processing_mock.c

```diff
@@ -140,46 +138,40 @@ int GoogleRtcAudioProcessingReconfigure(GoogleRtcAudioProcessingState *const sta
 	return 0;
 }
 
-int GoogleRtcAudioProcessingProcessCapture_int16(GoogleRtcAudioProcessingState *const state,
-						 const int16_t *const src,
-						 int16_t *const dest)
+int GoogleRtcAudioProcessingProcessCapture_float32(GoogleRtcAudioProcessingState *const state,
+						   const float *const *src,
+						   float * const *dest)
 {
-	int16_t *ref = state->aec_reference;
-	int16_t *mic = (int16_t *) src;
-	int16_t *out = dest;
-	int n, io, im, ir;
-
-	/* Mix input and reference channels to output. The matching channels numbers
-	 * are mixed. If e.g. microphone and output channels count is 4, and reference
-	 * has 2 channels, output channels 3 and 4 are copy of microphone channels 3 and 4,
-	 * and output channels 1 and 2 are sum of microphone and reference.
-	 */
-	memset(dest, 0, sizeof(int16_t) * state->num_output_channels * state->num_frames);
-	for (n = 0; n < state->num_frames; ++n) {
-		im = 0;
-		ir = 0;
-		for (io = 0; io < state->num_output_channels; io++) {
-			out[io] = sat_int16(
-				(im < state->num_capture_channels ? (int32_t)mic[im++] : 0) +
-				(ir < state->num_aec_reference_channels ? (int32_t)ref[ir++] : 0));
+	float *ref = state->aec_reference;
+	float **mic = (float **)src;
+	int n, chan, ref_chan;
+
+	for (chan = 0; chan < state->num_output_channels; chan++) {
+		for (n = 0; n < state->num_frames; ++n) {
+			float mic_save = mic[chan][n];	/* allow same in/out buffer */
+
+			if (chan < state->num_aec_reference_channels)
+				dest[chan][n] = mic_save + ref[n + (chan * state->num_frames)];
+			else
+				dest[chan][n] = mic_save;
```

Comment:
would move this if statement out of the second for loop be faster?
```
for (chan = 0; chan < state->num_output_channels; chan++) {
        if (chan < state->num_aec_reference_channels) 
		for (n = 0; n < state->num_frames; ++n) 
			dest[chan][n] = mic[chan][n] + ref[n + (chan * state->num_frames)];
       else
		for (n = 0; n < state->num_frames; ++n) 
			dest[chan][n] = mic[chan][n];
}
```

Reply:
...or even swapping the loops and splitting the one over channels into two loops:
```
for (n = 0; n < state->num_frames; ++n) {
    for (chan = 0; chan < state->num_aec_reference_channels; chan++)
        ...
    for (chan = state->num_aec_reference_channels; state->num_output_channels; chan++)
        ...
}
```

Reply:
this is mock, test only, nobody cares for performance ;)

Reply:
Agreed, mock is mostly for testing or bringup without the actual block, readability over performance here